File: user/src/main/java/com/top/shop/user/domain/RegisteredUser.java
package com.top.shop.user.domain;

import com.top.shop.user.util.Role;
import lombok.ToString;

import javax.persistence.*;
import java.util.List;

/**
 * @author Yome Mengistu
 */
@Entity
@ToString
public class RegisteredUser  extends User{

    @OneToOne(cascade = CascadeType.ALL)
    private UserAccount userAccount;


    public UserAccount getUserAccount() {
        return userAccount;
    }

    public void setUserAccount(UserAccount userAccount) {
        this.userAccount = userAccount;
        userAccount.setRole("REGISTERED_USER");
    }

}

File: Top-Shop-UI/src/app/models/userModel/RegisteredUser.ts
import {UserAccount} from "./UserAccount";
import {Address} from "./Address";
import {User} from "./User";
import {PaymentInformation} from "./PaymentInformation";

export class RegisteredUser extends User{
  private userAccount: UserAccount;


  constructor( role: string, address: Address[], paymentInformation: PaymentInformation[], user_role: string, userAccount: UserAccount) {
    super( role, address, paymentInformation, userAccount, user_role);
    this.userAccount = userAccount;
  }

}
